### PR TITLE
Various fixes for tests/CI

### PR DESCRIPTION
**Pull request by alvinhochun**
This PR contains some of the changes that will help adding the Clang CI (#1203) but are more about general fixes which are not specific to Clang.

### CODE REVIEW

**Review by kennykerr — APPROVED** [review]
Thanks!

File: test/test/multi_threaded_map.cpp [hunk 1]
```diff
@@ -63,7 +63,7 @@ namespace
         }
         else
         {
-            return static_cast<IObservableVector<K, V>>(winrt::make<custom_observable_map<K, V, Container>>(std::move(values)));
+            return static_cast<IObservableMap<K, V>>(winrt::make<custom_observable_map<K, V, Container>>(std::move(values)));
```
[the remark this answers is not in the document]
Comment (dunhor) [reply test/test/multi_threaded_map.cpp:66]: Hey, man. Sometimes I'm not the one to blame: https://github.com/microsoft/cppwinrt/commit/2cb852bbe163ff21ea5d9f898b1deece540c2100#diff-1bff145f07a25cc8dc26022e600d05a588e93e8114cd458e410a82c353ed916d

[the remark this answers is not in the document]
Comment (kennykerr) [reply test/test/multi_threaded_map.cpp:66]: @oldnewthing 👀

**Pull request merged**